### PR TITLE
WiFi Stack refactoring (utils and lwnl80211)

### PR DESCRIPTION
**Pull request by SeonghoByeon**
a86fccc network: change lwnl message handler 
- change lwnl message handler to socket to send events to multiple applications that request network event

cf4fc7d iotivity: enable log in iotivity sample program 
- print log in iotivity sample program to trace the events

86b1d2b iotivity: change the way to get network events in iotivity stack
- iotivity stack used message queue because there is no network event handler like netlink that linux provides. As lwnl has been implemented, iotivity is not no longer to use the message queue which make wifi_manager code contain unnecessary codes.

a405ff7 WiFi utils API refactoring
- Clean up configurations (Kconfig) and dependency
- Move Lwnl80211 Kconfig into WiFi Manager's

e99d184 os/drivers/wireless/scsc: Modify slsi api header directory
- Applying header location and refactoring
- Clean up scsc driver and slsi api code

dd6135d os/drivers/wireless/relatek: Modify rtk api header directory
- Applying header location and refactoring
- Clean up entire code

7526671 Remove CONFIG_NETUTILS_WIFI config
- CONFIG_NETUTILS_WIFI denotes the only SCSC WLAN driver for artik devices, not suitable for current WiFi Manager structure.

f784eb6 os/drivers/lwnl: Add scanning ops input
- Add scannig ops input to provide specific scan ap
- Bug fix: Passing the right scan_list when filtering is applied

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f784eb682f9bb58f123297c4f4e54f6c6e22f610] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f784eb682f9bb58f123297c4f4e54f6c6e22f610] - Code Rule Check (C++) OK.

File: apps/examples/iotivity_simpleserver/port_logger.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#define DEBUG_BUF_SIZE 128
+
+int ss_simple_logger(const char *fmt, ...)
+{
+	char str[DEBUG_BUF_SIZE];
+
+	va_list argp;
+	va_start( argp, fmt );
+    int ret = vsnprintf( str, DEBUG_BUF_SIZE, fmt, argp );
```
Comment (junmin-kim) [inline apps/examples/iotivity_simpleserver/port_logger.c]: This line use space.

File: framework/src/wifi_manager/wifi_event_monitor_internal.c [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+	}
+	g_cbk = *cbk;
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+
+int lwnl_leave_monitor(wifi_utils_cb_s *cb)
+{
+	(void)cb;
```
Comment (junmin-kim) [inline framework/src/wifi_manager/wifi_event_monitor_internal.c]: How about defining unused or add comment?

File: apps/examples/iotivity_simpleserver/port_logger.c (hunk 1 above)
Comment (SeonghoByeon) [reply apps/examples/iotivity_simpleserver/port_logger.c]: Modified. Thanks for the comment!

File: framework/src/wifi_manager/wifi_event_monitor_internal.c (hunk 2 above)
Comment (SeonghoByeon) [reply framework/src/wifi_manager/wifi_event_monitor_internal.c]: Attached a comment, please check it. Thanks!

**Comment by seinfra** [comment]
Target : [1220483ff567454991f6b31bc29b1d035e30a2c4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1220483ff567454991f6b31bc29b1d035e30a2c4] - Code Rule Check (C++) OK.

**Pull request merged**